log/log.go

```diff
@@ -0,0 +1,176 @@
+package log
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"os"
+	"runtime"
+
+	"github.com/almighty/almighty-core/configuration"
+
+	"github.com/goadesign/goa/middleware"
+	"golang.org/x/net/context"
+)
+
+var (
+	logger = NewLogger()
+)
+
+func NewLogger() *log.Logger {
+	logger := log.New()
+
+	if err := configuration.Setup(""); err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+
+	// TODO: This part should rely on a persistence configuration without having
+	// to call configuration.Setup(...)
+	if os.Getenv("ALMIGHTY_DEVELOPER_MODE_ENABLED") == "true" || configuration.IsPostgresDeveloperModeEnabled() {
+		customFormatter := new(log.TextFormatter)
+		customFormatter.FullTimestamp = true
+		customFormatter.TimestampFormat = "2006-01-02 15:04:05"
+		log.SetFormatter(customFormatter)
+
+		log.SetLevel(log.DebugLevel)
+		logger.Level = log.DebugLevel
+		logger.Formatter = customFormatter
+	} else {
+		customFormatter := new(log.JSONFormatter)
+		customFormatter.TimestampFormat = "2006-01-02 15:04:05"
+
+		log.SetFormatter(customFormatter)
+		customFormatter.DisableTimestamp = false
+
+		log.SetLevel(log.InfoLevel)
+		logger.Level = log.InfoLevel
+		logger.Formatter = customFormatter
+	}
+
+	logger.Out = os.Stdout
+
+	return logger
+}
+
+func NewCustomizedLogger(level string, defaultFormatter bool) (*log.Logger, error) {
+	logger := log.New()
+
+	lv, err := log.ParseLevel(level)
+	if err != nil {
+		return nil, err
+	}
+	logger.Level = lv
+
+	customFormatter := new(log.TextFormatter)
+	if !defaultFormatter {
+		customFormatter := new(log.JSONFormatter)
+		customFormatter.DisableTimestamp = false
+	}
+	customFormatter.TimestampFormat = "2006-01-02 15:04:05"
+	customFormatter.FullTimestamp = true
+	logger.Formatter = customFormatter
+
+	return logger, nil
+}
+
+func Logger() *log.Logger {
+	return logger
+}
+
+func LogError(ctx context.Context, fields map[string]interface{}, format string, args ...interface{}) {
```

Comment:
Why call the functions `LogError` and `LogWarn` an so forth instead of just `Error` and `Warn`. They are in the `log` package already which should make this obvious. I mean `log.Error(...)` is as clear as `log.LogError`, no?